examples/permissioned-token/test/common.js

```diff
@@ -0,0 +1,411 @@
+
+  getAssetHolding (address) {
+    return this.runtime.getAssetHolding(this.assetIndex, address);
+  }
+
+  // Opt-In account to ASA
+  optInToASA (address) {
+    this.runtime.optIntoASA(this.assetIndex, address, {});
+  }
+
+  // Opt-In address to Permissions SSC
+  optInToPermissionsSSC (address) {
+    this.runtime.optInToApp(address, this.permissionsAppId, {}, {});
+  }
+
+  issue (asaReserve, receiver, amount) {
+    issue(this.runtime, asaReserve, receiver, amount,
+      this.controllerAppID, this.assetIndex, this.lsig);
+  }
+
+  killToken (asaManager) {
+    killToken(this.runtime, asaManager, this.controllerAppID, this.assetIndex);
+  }
+
+  whitelist (permManager, addrToWhitelist) {
+    this.optInToPermissionsSSC(addrToWhitelist);
+    whitelist(this.runtime, permManager, addrToWhitelist, this.permissionsAppId);
+  }
+
+  transfer (from, to, amount) {
+    transfer(this.runtime, from, to, amount, this.assetIndex,
+      this.controllerAppID, this.permissionsAppId, this.lsig);
+  }
+
+  optOut (asaCreatorAddr, account) {
+    optOut(this.runtime, asaCreatorAddr, account, this.assetIndex);
+  }
+
+  forceTransfer (asaManager, from, to, amount) {
+    forceTransfer(this.runtime, asaManager, from, to, amount,
+      this.assetIndex, this.controllerAppID, this.permissionsAppId, this.lsig);
+  }
+}
+
+/**
+ * Issue some tokens to the passed account
+ * @param runtime RuntimeEnv Instance
+ * @param asaReserve ASA Reserve Account
+ * @param receiver Receiver Account
+ * @param amount Amount
+ * @param controllerAppID Controller App ID
+ * @param assetIndex ASA ID (Token ID)
+ * @param lsig (Clawback LSig)
+ */
+function issue (runtime, asaReserve, receiver, amount, controllerAppID, assetIndex, lsig) {
+  const txns = [
+    {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: asaReserve,
+      appId: controllerAppID,
+      payFlags: { totalFee: 1000 },
+      appArgs: ['str:issue'],
+      foreignAssets: [assetIndex]
+    },
+    {
+      type: types.TransactionType.RevokeAsset,
+      sign: types.SignType.LogicSignature,
+      fromAccountAddr: lsig.address(),
+      recipient: receiver.address,
+      assetID: assetIndex,
+      revocationTarget: asaReserve.addr,
+      amount: amount,
+      lsig: lsig,
+      payFlags: { totalFee: 1000 }
+    }
+  ];
+  runtime.executeTx(txns);
+}
+
+/**
+ * Kill the tokenfunction killToken (runtime, asaManager, controllerAppID) {
+ * @param runtime RuntimeEnv Instance
+ * @param asaManager ASA Manager Account
+ * @param controllerAppID Controller App ID
+ * @param assetIndex token index to be killed
+ */
+function killToken (runtime, asaManager, controllerAppID, assetIndex) {
+  runtime.executeTx({
+    type: types.TransactionType.CallNoOpSSC,
+    sign: types.SignType.SecretKey,
+    fromAccount: asaManager,
+    appId: controllerAppID,
+    payFlags: { totalFee: 1000 },
+    appArgs: ['str:kill'],
+    foreignAssets: [assetIndex]
+  });
+}
```

Comment:
I wonder if we could have same API and use it in scripts with algob and in tests with runtime - this  way we can define this functions once. Each of that functions will have a `txExecutor` or deployer (we would need a deployer test mode) to post transactions.

Reply:
We can think about it - i like the idea. It would be good for the user as well (using same high level API in scripts and in test).